### ISSUE TITLE
Port shaders to GLSL ES 3.0 (WebGL2 syntax)

### ISSUE THREAD

**Issue by birkskyum** [maplibre/maplibre-gl-js#2562]
Blocked by #2576 

There are approx. 60 shaders that have to be ported from GLSL 1 to 3.

Some shader work has happened in MapLibre Native as part of the preparation for a Metal backend where an old set of GL JS shades (before the terrain etc) were ported, but some testing indicate that adjustments to those shaders will be needed in order to use that work in GL JS, because there are subtle, but breaking, differences between desktop and web

**Comment by acalcutt** [comment]
Just wondering, does updating the shaders to ES 3.0 cause compatibility issues with macos like it did in native?

**Comment by birkskyum** [comment]
Yes, that is true. Apple did remove support for running opengl es 3.0 on desktop, but we can still use it here because we run it inside of the browser, so we only have to care about browser support which fortunately also counts safari now. Basically, web support for opengl es 3.0 has been lagging so many year behind desktop now (because of safari), that when it's now becoming sunset on desktop/mobile, the support number just recently became high enough to adopt it on web.

**Comment by birkskyum** [comment]
At an xxl I'd be happy to modernize or shaders. Would likely also bring a bit of performance, just because webgl2 does a few things smarter, but a bit UBO refactor is not part of the scope of this task

**Comment by HarelM** [comment]
I'll be happy to assign a XXL bounty on this, but I would like a few things before that please:
1. Better define an acceptance criteria
2. A design of what you are planning to do so we can after upon (can be split to a smaller bounty)
3. See how you can split the work into smaller pieces that can have separate, more smaller bounties (probably as part of 2)

This way you can better estimate how much time it would take and the project can receive smaller increments.

**Comment by birkskyum** [comment]
Acceptance criteria:

- all shaders are compiled with the '#version 300 es' flag set, indicating that it's GLSL ES 3.0.0

**Comment by birkskyum** [comment]
I bundled them to a single bounty because i know some will be here e than other, so it averaged it out a bit

**Comment by HarelM** [comment]
Assigning  XXL bounty.
It would be hard if you open a PR only when all is complete. But I can't think of a good way split this in terms of payment. So you'll need to decide how you want to do it. The bounty will be posted only when all shaders are converted and the code assumes version 3 for all shaders.

**Comment by birkskyum** [comment]
OC [link](https://opencollective.com/maplibre/expenses/140934)

**Comment by birkskyum** [comment]
@HarelM , I just realized there was never a direction linked to this bounty. The other tasks related to this was Modernize Codebase (1k/10k spent excl. this), so that would be most obvious. There is also the option for Performance direction (2k/10k spent excl. this).

**Comment by HarelM** [comment]
Yes, you are right.
It's related to #2576 basically, so same direction:
Bounty direction: https://github.com/maplibre/maplibre/issues/270
